substrate/frame/revive/src/tests/sol/terminate.rs

```diff
@@ -16,38 +16,60 @@
 // limitations under the License.
 
 use crate::{
-	test_utils::{builder::Contract, ALICE},
-	tests::{builder, ExtBuilder, Test},
-	Code, Config,
+	address::AddressMapper,
+	test_utils::{builder::Contract, ALICE, DJANGO, DJANGO_ADDR},
+	tests::{
+		builder,
+		test_utils::{get_balance, get_contract_checked},
+		Contracts, ExtBuilder, RuntimeOrigin, Test,
+	},
+	BalanceOf, Code, Config, Pallet, H160,
 };
 use alloy_core::sol_types::{SolCall, SolConstructor, SolValue};
 use frame_support::traits::fungible::Mutate;
 use hex_literal::hex;
-use pallet_revive_fixtures::{compile_module_with_type, FixtureType, Terminate};
+use pallet_revive_fixtures::{
+	compile_module_with_type, FixtureType, Terminate, TerminateCaller, TerminateDelegator,
+};
 use pretty_assertions::assert_eq;
-use test_case::test_case;
+use test_case::{test_case, test_matrix};
 
 /// Decode a contract return value into an error string.
 fn decode_error(output: &[u8]) -> String {
 	assert!(output.len() >= 4 && &output[..4] == &hex!("08c379a0"));
 	String::abi_decode(&output[4..]).unwrap()
 }
 
-#[test_case(FixtureType::Solc)]
-#[test_case(FixtureType::Resolc)]
-fn base_case(fixture_type: FixtureType) {
+const METHOD_PRECOMPILE: u8 = 0;
```

Comment:
Solidity has an `enum` type which seems to fit the use case here. But eitherways is fine for me.